rpcs3/Emu/Cell/Modules/cellAudioOut.cpp

```diff
@@ -187,28 +366,71 @@ error_code cellAudioOutConfigure(u32 audioOut, vm::ptr<CellAudioOutConfiguration
 	switch (audioOut)
 	{
 	case CELL_AUDIO_OUT_PRIMARY:
+		break;
+	case CELL_AUDIO_OUT_SECONDARY:
+		return CELL_AUDIO_OUT_ERROR_UNSUPPORTED_AUDIO_OUT;
+	default:
+		return CELL_AUDIO_OUT_ERROR_ILLEGAL_PARAMETER;
+	}
+
+	audio_out_configuration::audio_out out_old;
+	audio_out_configuration::audio_out out_new;
+
+	audio_out_configuration& cfg = g_fxo->get<audio_out_configuration>();
 	{
-		if (config->channel)
-		{
-			//Emu.GetAudioManager().GetInfo().mode.channel = config->channel;
-		}
+		std::lock_guard lock(cfg.mtx);
 
-		//Emu.GetAudioManager().GetInfo().mode.encoder = config->encoder;
+		audio_out_configuration::audio_out& out = cfg.out.at(audioOut);
 
-		if (config->downMixer)
+		if (out.sound_modes.cend() == std::find_if(out.sound_modes.cbegin(), out.sound_modes.cend(), [&config](const CellAudioOutSoundMode& mode)
+			{
+				return mode.channel == config->channel && mode.type == config->encoder && config->downMixer <= CELL_AUDIO_OUT_DOWNMIXER_TYPE_B;
+			}))
 		{
-			//Emu.GetAudioManager().GetInfo().mode.downMixer = config->downMixer;
+			return CELL_AUDIO_OUT_ERROR_ILLEGAL_CONFIGURATION; // TODO: confirm
 		}
 
-		return CELL_OK;
+		out_old = out;
+
+		out.channels = config->channel;
+		out.encoder = config->encoder;
+		out.downmixer = config->downMixer;
+
+		out_new = out;
 	}
 
-	case CELL_AUDIO_OUT_SECONDARY:
-		return CELL_AUDIO_OUT_ERROR_UNSUPPORTED_AUDIO_OUT;
-	default: break;
+	if (std::memcmp(&out_old, &out_new, sizeof(audio_out_configuration::audio_out)) != 0)
```

Comment:
You can't memcmp 2 structs with a std::vector in them for equality.
The vector contains a heap address and even if it contains the same values the comparison will come back not equal.

Reply:
It was old code from before I added vectors.
Will fix later.